functional/sev/run.sh

```diff
@@ -0,0 +1,444 @@
+
+export GOROOT="/usr/local/go"
+export GOPATH="${WORKING_DIR}/go"
+export PATH="${PATH}:${GOROOT}/bin"
+
+export KBS_DB_HOST="127.0.0.1"
+export KBS_DB_USER="root"
+export KBS_DB_PW="root"
+export KBS_DB="simple_kbs"
+export KBS_DB_TYPE="mysql"
+
+esudo() {
+  sudo -E PATH=$PATH "$@"
+}
+
+build_and_install_kata_runtime_4270() {
+  cd ${WORKING_DIR}
+
+  # Clone kata-containers repo and checkout PR 4270
+  git clone https://github.com/kata-containers/kata-containers.git
+  cd kata-containers
+  gh pr checkout 4270
+
+  # Build and install the CC Kata Containers runtime
+  export GO111MODULE=on
+  cd ${WORKING_DIR}/kata-containers/src/runtime
+  make && esudo make install
+}
+
+build_rootfs() {
+  cd ${WORKING_DIR}/kata-containers/tools/osbuilder/rootfs-builder
+  esudo GOPATH=$GOPATH AGENT_INIT=yes USE_DOCKER=true DOCKER_RUNTIME=runc AA_KBC=offline_sev_kbc SKOPEO_UMOCI=yes ./rootfs.sh ubuntu
+
+esudo tee rootfs-ubuntu/etc/default/agent.cfg <<EOF
+aa_kbc_params = "offline_sev_kbc::null"
+[endpoints]
+allowed = [
+"AddARPNeighborsRequest",
+"AddSwapRequest",
+"CloseStdinRequest",
+"CopyFileRequest",
+"CreateContainerRequest",
+"CreateSandboxRequest",
+"DestroySandboxRequest",
+"GetMetricsRequest",
+"GetOOMEventRequest",
+"GuestDetailsRequest",
+"ListInterfacesRequest",
+"ListRoutesRequest",
+"MemHotplugByProbeRequest",
+"OnlineCPUMemRequest",
+"PauseContainerRequest",
+"PullImageRequest",
+"ReadStreamRequest",
+"RemoveContainerRequest",
+"ResumeContainerRequest",
+"SetGuestDateTimeRequest",
+"SignalProcessRequest",
+"StartContainerRequest",
+"StartTracingRequest",
+"StatsContainerRequest",
+"StopTracingRequest",
+"TtyWinResizeRequest",
+"UpdateContainerRequest",
+"UpdateInterfaceRequest",
+"UpdateRoutesRequest",
+"WaitProcessRequest",
+"WriteStreamRequest"
+]
+EOF
+}
+
+build_kernel() {
+  cd ${WORKING_DIR}/kata-containers/tools/packaging/kernel
+
+  # Setup and generate kernel .config
+  ./build-kernel.sh -x sev setup
+
+  # CONFIG_EFI_STUB - can be removed when kernel upgraded to 5.19
+  (
+    cd kata-linux-efi-secret-*
+    ./scripts/config --enable CONFIG_EFI_STUB
+    make olddefconfig
+  )
+
+  # Build and copy modules to previously generated rootfs
+  ./build-kernel.sh -x sev build
+  esudo cp -r kata-linux-efi-secret-*/lib/modules ${WORKING_DIR}/kata-containers/tools/osbuilder/rootfs-builder/rootfs-ubuntu/lib/
+
+  # Can't run this as non-sudo, need root access to copy files to /usr/share/kata-containers
+  # Can't run this as sudo (or esudo, alias above) because 'sudo yq' gives 'permission denied error'
+  # Install
+  #esudo ./build-kernel.sh -x sev install
+
+  # Per above, run this to install
+  bzImage="kata-linux-efi-secret-v5.17-rc6-93/arch/x86_64/boot/bzImage"
+  vmlinuz="/usr/share/kata-containers/vmlinuz-efi-secret-v5.17-rc6-93-sev"
+  esudo install -o root -g root -m 0644 -D "${bzImage}" "${vmlinuz}"
+  esudo ln -sf "${vmlinuz}" "/usr/share/kata-containers/vmlinuz-sev.container"
+}
+
+build_initrd() {
```

Comment:
As I mentioned in my comments to `build_rootfs()`, we should be able to remove this rebuild of the guest image. There is only one thing I'd like to check, here you export `AGENT_INIT=yes` and I guess it is the default already.

@stevenhorsman the guest image for CC is already built with `AGENT_INIT=yes`, doesn't it?

Reply:
In CC at the moment we just use the default, which is `AGENT_INIT=no`, so I guess that would need to go in the ci_job_flags parameters for this job?

Reply:
@ryansavino do we really need the agent as init process? If so, then we will have to build a guest image specifically to SEV

Reply:
SEV requires initrd image. I have fixes for initrd with systemd in https://github.com/kata-containers/kata-containers/pull/4987. Need this PR to get in to proceed with testing. I also have changes in `.ci/install_kata_image.sh` to accommodate as well.